File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/setshapeappearance.rb
module Asposediagramjava
  module SetShapeAppearance
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Access a particular page
        page = diagram.getPages().getPage("Flow 1")

        # get a particular connector shape
        shape = page.getShapes().getShape(1)
        
        # set dynamic connector appearance
        shape.setConnectorsType(Rjb::import('com.aspose.diagram.ConnectorsTypeValue').STRAIGHT_LINES)

        # Save diagram
        diagram.save(data_dir + "SetShapeAppearance.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Set appearnce of the connector type shape."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/SetShapeAppearance.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
use com\aspose\diagram\ConnectorsTypeValue as ConnectorsTypeValue;
class SetShapeAppearance{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram =new Diagram($dataDir."Drawing.vsd");

        # Access a particular page
        $page=$diagram->getPages()->getPage("Flow 1");

        # get a particular connector shape
        $shape=$page->getShapes()->getShape(1);

        # set dynamic connector appearance
        $connectorsTypeValue=new ConnectorsTypeValue();
        $shape->setConnectorsType($connectorsTypeValue->STRAIGHT_LINES);

        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."SetShapeAppearance.vdx",$saveFileFormat->VDX);
        print "Set appearnce of the connector type shape.".PHP_EOL;
    }
}